index.php

```diff
@@ -203,3 +209,58 @@ function strip_header_and_footer($lines, $show_intro){
 		return $stripped_lines;
 	}
 
+    #
+    # Handle save and restore for later
+    #
+    function handle_save_and_restore($command, $save_path, $session_id, $FROTZ_SLOT_SAVES_ENABLED, $FROTZ_SAVE_SLOTS, $FROTZ_SAVE_PATH)
+    {
+        if (!$FROTZ_SLOT_SAVES_ENABLED) {
+            $data = array(
+                'title' => 'Game save',
+                'message' => "Sorry, slot-based saving is not enabled.",
+            );
+            $ret = handler_output($data);
+            exit();
+        }
+
+        if ($FROTZ_SLOT_SAVES_ENABLED && preg_match('/(?:(save)\\s+(\\d))/i', $command, $save_slot)) {
+
+            if ($save_slot <= $FROTZ_SAVE_SLOTS && $save_slot > 0) {
+                copy($save_path, "{$FROTZ_SAVE_PATH}/{$session_id}-${save_slot[2]}.zsav");
```

Comment:
Is there a reason some of these are `${save_slot...` as oppose to `{$save_slot...`? I have yet to encounter that syntax.